RenderStatePackager/src/RenderStatePackager.cpp

```diff
@@ -25,12 +25,156 @@
  */
 
 #include "RenderStatePackager.hpp"
+#include "GraphicsAccessories.hpp"
+#include "BasicMath.hpp"
 #include "DefaultRawMemoryAllocator.hpp"
 #include "DynamicLinearAllocator.hpp"
+#include "SerializedPipelineState.h"
+#include "FileSystem.hpp"
+#include "FileWrapper.hpp"
 
 namespace Diligent
 {
 
+struct BytecodeDumper
+{
+    class WorkingDirectory
+    {
+    public:
+        WorkingDirectory(const Char* Path) :
+            m_CurrentPath{Path} {}
+
+
+        void Push(const Char* Path)
+        {
+            m_CurrentPath = m_CurrentPath + FileSystem::GetSlashSymbol() + Path;
+
+            if (!FileSystem::PathExists(m_CurrentPath.c_str()))
+                FileSystem::CreateDirectory(m_CurrentPath.c_str());
+        }
+
+        void Pop()
+        {
+            m_CurrentPath = m_CurrentPath.substr(0, m_CurrentPath.find_last_of((FileSystem::GetSlashSymbol())));
+        }
+
+        String ComputePathFor(const Char* FileName) const
+        {
+            return m_CurrentPath + FileSystem::GetSlashSymbol() + FileName;
+        }
+
+    private:
+        String m_CurrentPath;
+    };
+
+    class WorkingDirectoryScope
+    {
+    public:
+        WorkingDirectoryScope(WorkingDirectory& Directory, const Char* Path) :
+            m_Directory{Directory}
+        {
+            m_Directory.Push(Path);
+        }
+
+        ~WorkingDirectoryScope()
+        {
+            m_Directory.Pop();
+        }
+
+    private:
+        WorkingDirectory& m_Directory;
+    };
+
+
+    static bool Execute(const std::vector<RefCntAutoPtr<IPipelineState>>& Pipelines, ARCHIVE_DEVICE_DATA_FLAGS DeviceFlags, const char* Path)
+    {
+        auto GetFileExtension = [](ARCHIVE_DEVICE_DATA_FLAGS DeviceFlag, SHADER_SOURCE_LANGUAGE Language, bool UseBytecode) //
```

Comment:
No need to overuse lambdas. This is a long function and it is natural and more clear to make it a private member